### PR TITLE
Add QueryResponse::is_success API and remove debug impls

### LINKED ISSUE #110: Fix Debug implementation for SequenceRead
The debug implementation of Sequence read is missing important debugging information

### PR DESCRIPTION
**Pull request by lionel-faber**
closes #110

### CODE REVIEW

File: src/client/mod.rs [hunk 1]
```diff
@@ -299,6 +298,39 @@ pub enum QueryResponse {
...
 
+impl QueryResponse {
+    /// Returns true if the result returned is a success or not
+    pub fn is_success(&self) -> bool {
+        use QueryResponse::*;
+        match self {
+            GetBlob(result) => result.is_ok(),
+            GetMap(result) => result.is_ok(),
+            GetMapShell(result) => result.is_ok(),
+            GetMapVersion(result) => result.is_ok(),
```
Comment (Yoga07) [inline src/client/mod.rs:309]: Just thinking if there is a better way to do this? 🤔

**Review by Yoga07 — DISMISSED** [review]
LGTM

File: src/client/register.rs [hunk 2]
```diff
@@ -124,10 +108,10 @@ impl fmt::Debug for RegisterWrite {
         write!(
             formatter,
             "RegisterWrite::{}",
-            match *self {
-                RegisterWrite::New(_) => "NewRegister",
-                RegisterWrite::Delete(_) => "DeleteRegister",
-                RegisterWrite::Edit(_) => "EditRegister",
+            match self {
+                RegisterWrite::New(register) => format!("New({:?})", register.address()),
```
Comment (bochaco) [inline src/client/register.rs:112]: Perhaps we want `NewRegister({:?})` not only to keep consistency with `DeleteRegister({:?})` but also to see what's new/edited?

File: src/client/mod.rs (hunk 1 above)
Comment (bochaco) [reply src/client/mod.rs:309]: I assume grouping them with `|` is not allowed by the compiler due to the types? I mean `GetMap(result) | GetMapShell(result) | ... | GetMapVersion(result) => result.is_ok()`

File: src/client/register.rs (hunk 2 above)
Comment (lionel-faber) [reply src/client/register.rs:112]: Delete should be just "Delete" I can update that. 

The problem with writing the whole register is that the log will become too verbose. In the context of the nodes we need not know what the content of the new register is right? That can be in the layer where data operations are performed?

File: src/client/mod.rs (hunk 1 above)
Comment (lionel-faber) [reply src/client/mod.rs:309]: ~I assumed that as well. I just made the change and it works :) Will update~
It doesn't. Initial assumption was right.

File: src/client/register.rs (hunk 2 above)
Comment (bochaco) [reply src/client/register.rs:112]: ok, yeah i guess it depends who is logging this, i just thought that if you blindly log it you don't know if it's a New Register or new what...

Comment (lionel-faber) [reply src/client/register.rs:112]: Yeah. In this context it will always be `RegisterWrite::New(register_address)`

Comment (bochaco) [reply src/client/register.rs:112]: ok, we can change it if it's not good enough in the future. But that's what I mean, you won't get `RegisterWrite::New(register_address)` written to the log, but just `New(register_address)` ?

**Review by bochaco — APPROVED** [review]

File: src/client/register.rs (hunk 2 above)
Comment (lionel-faber) [reply src/client/register.rs:112]: `RegisterWrite` is mentioned in L110

Comment (bochaco) [reply src/client/register.rs:112]: ahhhh!!! I was missing that completely! sorry about that!

**Pull request merged**